src/optimization/Eliminator.cpp

```diff
@@ -363,7 +393,18 @@ void optimizations::eliminateRedundantMoves(const Module& module, Method& method
 			auto sourceWriter = (move->getSource().getSingleWriter() != nullptr) ? it.getBasicBlock()->findWalkerForInstruction(move->getSource().getSingleWriter(), it) : Optional<InstructionWalker>{};
 			auto destinationReader = (move->hasValueType(ValueType::LOCAL) && move->getOutput()->local->getUsers(LocalUse::Type::READER).size() == 1) ? it.getBasicBlock()->findWalkerForInstruction(*move->getOutput()->local->getUsers(LocalUse::Type::READER).begin(), it.getBasicBlock()->end()) : Optional<InstructionWalker>{};
 
-			if(!it->hasSideEffects() && sourceUsedOnce && destUsedOnce && destinationReader && move->getSource().type == move->getOutput()->type)
+			if(! move->hasPackMode() && ! move->hasUnpackMode() && move->getSource() == move->getOutput().value())
+			{
+				if (move->signal != SIGNAL_NONE)
```

Comment:
Why do you erase the move when it triggers a signal? The signal will get lost!
Shouldn't you only remove the move if it has no side-effects, e.g. by checking `!move->hasSideEffects()`?